azurelinuxagent/ga/update.py

```diff
@@ -481,13 +482,16 @@ def _try_update_goal_state(self, protocol):
         Attempts to update the goal state and returns True on success or False on failure, sending telemetry events about the failures.
         """
         try:
-            protocol.update_goal_state()
+            max_errors_to_log = 3
+
```

Comment:
As part of the Fast Track changes more logging has been added to the GoalState class. I added the capability to turn off all logging done by GoalState.
 
We leaving logging turned on for at most 3 consecutive errors and then turn it off and output 1 single error every 6 hours. We turn on logging once fetching goal state succeeds.

Reply:
@narrieta Sorry, I didn't get a chance to review this. Do we turn on before the goal state succeeded or will it be for next goal state? Looks to me successful goal state may not have logging if it gets updated after 3 consecutive errors. 

Ex: We had 3 consecutive failures, on 4th attempt we disable the logger but goal state updated successfully and process the same goal state. So don't we miss that process goal state log in waagent because logger disabled.

Reply:
thanks for taking a look @nagworld9 

yes, we re-enable logging at line 492

Reply:
(we reset to 0 and the condition for turning it off is "self._update_goal_state_error_count >= max_errors_to_log")